### PR TITLE
refactor PeerCommand to use PicoCLI exceptions

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

avoid the stack trace

see #977

### CODE REVIEW

**Review by rolfyone — APPROVED** [review]
LGTM

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/PeerCommand.java [hunk 1]
```diff
@@ -68,14 +81,22 @@ public void generate(
           int number)
       throws IOException {
     try {
-      File f = new File(params.outputFile);
+      validateParamsAndGenerate(params.outputFile, number);
+    } catch (Exception ex) {
```
Comment (usmansaleem) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/PeerCommand.java]: nit: `final Exception`

File: teku/src/test/java/tech/pegasys/teku/cli/subcommand/PeerCommandTest.java [hunk 2]
```diff
@@ -19,19 +19,18 @@
...
-import tech.pegasys.teku.cli.subcommand.PeerCommand.PeerGenerationParams;
 import tech.pegasys.teku.util.config.InvalidConfigurationException;
 
 public class PeerCommandTest {
 
   @Test
   public void peerGenerate_shouldDisplayErrorIfFileNotFound() {
     PeerCommand peerCommand = new PeerCommand();
-    PeerGenerationParams params = new PeerGenerationParams("directory-does-not-exist/config.dat");
+    String fileWhichDoesNotExist = "directory-does-not-exist/config.dat";
```
Comment (usmansaleem) [inline teku/src/test/java/tech/pegasys/teku/cli/subcommand/PeerCommandTest.java]: nit: final, unless it needs to be overwritten later on.

**Review by usmansaleem — APPROVED** [review]
LGTM, minor suggestions around using final keyword.

File: teku/src/test/java/tech/pegasys/teku/cli/subcommand/PeerCommandTest.java (hunk 2 above)
Comment (macfarla) [reply teku/src/test/java/tech/pegasys/teku/cli/subcommand/PeerCommandTest.java]: done

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/PeerCommand.java (hunk 1 above)
Comment (macfarla) [reply teku/src/main/java/tech/pegasys/teku/cli/subcommand/PeerCommand.java]: done

**Review by usmansaleem — APPROVED** [review]
LGTM

**Pull request merged**